docs/docs/aleph_zero/sdk.md

```diff
@@ -0,0 +1,413 @@
+```
+
+### Open position
+
+```typescript
+// token y has 12 decimals and we want to add 8 actual tokens to our position
+const tokenYAmount = 8n * 10n ** 12n
+
+// set lower and upper tick, we want to create position in range [-10, 10]
+const lowerTick = -10n
+const upperTick = 10n
+
+const liquidityResult = getLiquidityByX(tokenYAmount, lowerTick, upperTick, initSqrtPrice, true)
+const tokenXAmount = liquidityResult.amount
+const positionLiquidity = liquidityResult.l
+
+// print amount of token x and y we need to give to open position based on parameteres we passed
+console.log(tokenYAmount, tokenXAmount)
+
+// approve transfers of both tokens
+await psp22.approve(
+  account,
+  invariant.contract.address.toString(),
+  isTokenX(TOKEN0_ADDRESS, TOKEN1_ADDRESS) ? tokenYAmount : tokenXAmount
+)
+await psp22.setContractAddress(TOKEN1_ADDRESS)
+await psp22.approve(
+  account,
+  invariant.contract.address.toString(),
+  isTokenX(TOKEN0_ADDRESS, TOKEN1_ADDRESS) ? tokenXAmount : tokenYAmount
+)
+await psp22.setContractAddress(TOKEN0_ADDRESS)
+
+// open up position
+await invariant.createPosition(
+  account,
+  poolKey,
+  lowerTick,
+  upperTick,
+  positionLiquidity,
+  initSqrtPrice,
+  initSqrtPrice
+)
+```
+
+### Perform swap
+
+:::info How to calculate input amount
+In order to calculate input amount, we have to multiply actual token amount you want to swap times 10 to the power of decimal.
+Let's say some token has decimal of 12 and we want to swap 6 actual tokens. Here is how we can calculate input amount:
+6 \* 10^12 = 6000000000000.
+:::
+
+```typescript
+// --snip--
+
+const poolKey = newPoolKey(TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+
+// make sure that you swap in correct direction,
+// here we check if token0 (token we want to give) is tokenX
+const xToY = isTokenX(TOKEN0_ADDRESS, TOKEN1_ADDRESS)
+
+// here we want to swap 6 token0
+// token0 has 12 decimals so we need to multiply it by 10^12
+const amount = 6n * 10n ** 12n
+
+// approve token0 transfer
+await psp22.approve(account, invariant.contract.address.toString(), amount)
+
+// slippage is a price change you are willing to accept,
+// for examples if current price is 1 and your slippage is 1%, then price limit will be 1.01
+const allowedSlippage = toPercentage(1n, 3n) // 0.001 = 0.1%
+
+// swap message only accepts price limit so we have to calculate it in order to use slippage,
+// here we are passing current price and allowed slippage to calculate price limit
+const pool = await invariant.getPool(account, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+const sqrtPriceLimit = calculateSqrtPriceAfterSlippage(pool.sqrtPrice, allowedSlippage, !xToY)
+
+const result = await invariant.swap(account, poolKey, xToY, amount, true, sqrtPriceLimit)
+console.log(result.hash) // print transaction hash
+```
+
+### Query state
+
+```typescript
+// --snip--
+
+// query state
+const poolAfter = await invariant.getPool(account, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
```

Comment:
It will be better to mix the query here with the results struct. E.g. `invariant.getPool` and `Pool`